publisher/internal/analytics/ethereum/operations.go

```diff
@@ -284,6 +320,23 @@ func (add *Addition) handleLiquidityTransfer(mint EventLog, transfer EventLog) {
 	}
 }
 
+func (rem *Removal) calculateFeesEarned(collectLog EventLog) error {
+	_, token0HexAmount, token1HexAmount, err := splitCollectDatatoHexFields(collectLog.Data) // Token order original as in liquidity pool
+	if err != nil {
+		return err
+	}
+
+	rem.Token0Earned, rem.Token1Earned = rem.Token0, rem.Token1
+	rem.Token0Earned.Amount = convertTransferAmount(token0HexAmount, rem.Token0.Decimals) - rem.Token0.Amount
+	rem.Token1Earned.Amount = convertTransferAmount(token1HexAmount, rem.Token1.Decimals) - rem.Token1.Amount
+
+	if !rem.Position.isOrderCorrect() {
```

Comment:
Seems I cannot comment on unmodified lines, so comment here:
`isOrderCorrect` seems a bit too domain-specific, so it would be a little clearer if the name reflected what actually it checks for.